app/forms/candidate_interface/degree_subject_form.rb

```diff
@@ -23,7 +23,18 @@ def assign_form_values
   private
 
     def hesa_code
-      Hesa::Subject.find_by_name(subject)&.hesa_code
+      degree_subject&.hesa_code
+    end
+
+    def degree_subject_uuid
+      new_data_degree_subject = DfE::ReferenceData::Degrees::SUBJECTS.all.find do |new_data_subject|
+        new_data_subject.name == subject
```

Comment:
Should we check synonyms here too?

Reply:
The degree subject is the only one that it can find all in old and new data with equivalency and the synonyms are blank in the new data. But maybe we should do it when we replace the old data in my next PR. 🤔